common/maplib/move-rule.mjs

```diff
@@ -0,0 +1,51 @@
+import {createRequire} from 'module';
+const require = createRequire(import.meta.url);
+
+const config = require('config');
+const movingRequestThreshold = config.get('movingRequestThreshold');
+
+class MoveRule{
+
+  _getManhattanDistance(a,b){
+    return Math.abs(a.x - b.x) + Math.abs(a.y - b.y);
+  }
+
+  sameMapCheck(mapCoord1,mapCoord2){
+      return mapCoord1.mapName === mapCoord2.mapName;
+  }
+
+  borderCheck(coord,mapSize){
+    if (coord.x >= mapSize.width || coord.x < 0) {
+      return false;
+    }
+    if (coord.y >= mapSize.height || coord.y < 0) {
+      return false;
+    }
+    return true;
+  }
+
+  nearbyGridCheck(coord,lastCoord){
```

Comment:
I still don't know why we need this function after we have a speed check.
Speed check already contains the functionality of what you want, doesn't it?

Reply:
Okay, I think I know what you are thinking.
I would prefer checking nearby grid in speed check function.
**Speed** is related to both **distance** and **time**.